Firestore/core/src/firebase/firestore/util/status.cc

```diff
@@ -28,9 +28,8 @@ namespace util {
 
 Status::Status(Error code, absl::string_view msg) {
   HARD_ASSERT(code != Error::Ok);
-  state_ = absl::make_unique<State>();
-  state_->code = code;
-  state_->msg = static_cast<std::string>(msg);
+  state_ = MakeStatePtr(
+      Status::State(std::move(code), static_cast<std::string>(msg)));
```

Comment:
Using the variadic overload, this can be just
```cpp
state_ = MakeStatePtr(code, msg);
```
I.e., just forward the arguments to `State` constructor, without calling the constructor yourself.